src/supervisor/watchers/handlers/index.ts

```diff
@@ -109,6 +110,20 @@ export function setupInformer(namespace: string, workloadKind: WorkloadKind) {
 
   const informer = makeInformer<KubernetesObject>(kubeConfig, namespacedEndpoint, loggedListMethod);
 
+  informer.on(ERROR, (err) => {
+    // Types from client library insists that callback is of type KubernetesObject
+    if ((err as any).code === ECONNRESET_ERROR_CODE) {
+      logger.debug('informer ECONNRESET occurred, restarting informer');
+
+      // Restart informer after 1sec
+      setTimeout(() => {
+        informer.start();
+      }, 1000);
+    } else {
+      logger.error({ err }, 'unexpected informer error event occurred');
```

Comment:
Should we also throw to kill the program? Or we don't do it so that we don't lose the logs?

Reply:
We ideally shouldn't get to here, so it would be nice to have logs if smth happens, so we can act on that.
You can always have logs from previous pod by doing `kc logs <pod-id> --previos`